dds/DCPS/XTypes/TypeLookupService.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include "DCPS/DdsDcps_pch.h"
+#include "TypeLookupService.h"
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace XTypes {
+
+TypeLookupService::TypeLookupService()
+{
+  to_empty_.minimal.kind = TK_NONE;
+  to_empty_.complete.kind = TK_NONE;
+  rpc_sequence_number_ = 0;
+}
+
+
+TypeLookupService::~TypeLookupService()
+{
+}
+
+
+void TypeLookupService::get_type_objects(const TypeIdentifierSeq& type_ids,
+                                         TypeIdentifierTypeObjectPairSeq& types)
+{
+  for (CORBA::ULong i = 0; i < type_ids.length(); ++i) {
+    TypeObjectMap::iterator it_object = type_object_map_.find(type_ids[i]);
+    if (it_object != type_object_map_.end()) {
+      TypeIdentifierTypeObjectPair new_to_pair(it_object->first, it_object->second);
+      types.append(new_to_pair);
+    }
+  }
+}
+
+
+const TypeObject& TypeLookupService::get_type_objects(const TypeIdentifier& type_id) const
+{
+  TypeObjectMap::const_iterator it_object = type_object_map_.find(type_id);
+  if (it_object != type_object_map_.end()) {
+    return it_object->second;
+  }
+  return to_empty_;
+}
+
+
+void TypeLookupService::add_type_objects_to_cache(TypeIdentifierTypeObjectPairSeq& types)
+{
+  for (ACE_UINT32 i = 0; i < types.length(); ++i) {
+    TypeObjectMap::iterator it_type_id_with_size_seq = type_object_map_.find(types[i].type_identifier);
```

Comment:
```suggestion
    const TypeObjectMap::iterator it_type_id_with_size_seq = type_object_map_.find(types[i].type_identifier);
```
I'm going to stop looking for these and let you take over at this point